clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/provider/view/KubernetesJobProvider.java

```diff
@@ -73,14 +76,32 @@ public KubernetesJobStatus collectJob(String account, String location, String id
             jobStatus.getLocation(),
             KubernetesSelectorList.fromMatchLabels(selector));
 
-    jobStatus.setPods(
+    List<V1Pod> typedPods =
         pods.stream()
-            .map(
-                p -> {
-                  V1Pod pod = KubernetesCacheDataConverter.getResource(p, V1Pod.class);
-                  return new KubernetesJobStatus.PodStatus(pod);
+            .map(m -> KubernetesCacheDataConverter.getResource(m, V1Pod.class))
+            .sorted(
+                (p1, p2) -> {
+                  DateTime dtDefault = new DateTime(0);
+                  DateTime time1 =
+                      p1.getStatus() != null
+                          ? Optional.ofNullable(p1.getStatus().getStartTime()).orElse(dtDefault)
+                          : dtDefault;
+                  DateTime time2 =
+                      p2.getStatus() != null
+                          ? Optional.ofNullable(p2.getStatus().getStartTime()).orElse(dtDefault)
+                          : dtDefault;
+                  return time1.compareTo(time2);
                 })
-            .collect(Collectors.toList()));
+            .collect(Collectors.toList());
+
+    V1Pod mostRecentPod = typedPods.get(typedPods.size() - 1);
+    jobStatus.setMostRecentPodName(
+        mostRecentPod.getMetadata() != null ? mostRecentPod.getMetadata().getName() : "");
+
+    if (detailedPodStatus) {
+      jobStatus.setPods(
```

Comment:
Does `KubernetesJobStatus` serialize with `pods` if null?

Reply:
`pods` are not included in the response